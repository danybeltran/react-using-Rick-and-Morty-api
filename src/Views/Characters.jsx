import React, { useState, useEffect, useContext } from "react";
import { useParams, Link } from "react-router-dom";
import { Character } from "../components/Character";
import { AppContext } from "../Context";
import { SearchButton } from "../components/CharactersView/SearchButton";
import { SearchInput } from "../components/CharactersView/SearchInput";
import { Pagination } from "../components/CharactersView/Pagination";
import { MappedCharacters } from "../components/CharactersView/MappedCharacters";

export const Characters = () => {

    const context = useContext(AppContext);

    const {
        characters,
        setPage,
        setSearchQuery,
        userWillSearch,
        setUserWillSearch
    } = useContext(AppContext);

    const { page } = useParams();

    useEffect(() => {
        setPage(page);
    }, [page])

    return (
        <div className="characters">
            <h3 className="characters-header">All characters</h3>
            <br />
            <div className="characters-body">
                <Pagination />
            </div>
            <div className="characters-result">
                <MappedCharacters />
                {
                    // If the search gave no results
                    (characters.length === 0 && userWillSearch) && (
                        <div className="zero-found">
                            <h2 className="text-center">No results</h2>
                        </div>
                    )
                }
            </div>
        </div>
    )
}