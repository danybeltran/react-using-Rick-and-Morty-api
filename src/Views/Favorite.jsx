<<<<<<< HEAD
import React, { useState, useEffect } from "react";
import { Character } from "../components/Character";
import MortyFace from "../img/morty-no-favorites.png";

export const Favorite = () => {
    // get the string of all ids separated by comma to fetch the api
    const [AllFavoriteCharactersId] = useState(localStorage["favorite"].replace(/\[|\]/g, ""));

    const [favoriteCharacters, setFavoriteCharacters] = useState([]);


    useEffect(() => {
        const getFavoriteCharacters = async (ids) => {
            var favorite = await fetch("https://rickandmortyapi.com/api/character/" + ids + ",");
            favorite = await favorite.json();
            setFavoriteCharacters(favorite);
        }
        getFavoriteCharacters(AllFavoriteCharactersId);
    },[AllFavoriteCharactersId])
    return (
        <div className="favorite">
            <h3 className="characters-header">Your favorite characters</h3>
            <br />
            <div className="favorite-b">
                {
                    typeof favoriteCharacters.map !== "undefined" ?
                        favoriteCharacters.map((character) => <Character props={character} isFavorite={true} key={character.id} />)
                        :
                        <>
                            <div className="container-no-results">
                                <img src={MortyFace} className="no-results-image" alt="" />
                                <h3 className="no-results-text">List is empty</h3>
                            </div>
                        </>
                }
            </div>
        </div>
    )
=======
import React from "react";
import { FavoriteContainer } from "../components/Favorite/FavoriteContainer";
import { FavoriteCharactersHeader } from "../components/Favorite/FavoriteCharactersHeader";
import { FavoriteCharacters } from "../components/Favorite/FavoriteCharacters";

export const Favorite = () => {
    return (
        <FavoriteContainer>
            <FavoriteCharactersHeader text="Your favorite characters" />
            <br />
            <FavoriteCharacters />
        </FavoriteContainer>
    )
>>>>>>> 32ef5e00d4735d6783d785b48d2330430e388d02
}