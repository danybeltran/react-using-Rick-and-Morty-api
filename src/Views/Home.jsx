import React from "react";
import RickAndMortyTogether from "../img/rick-and-morty-together.png";
import { Link } from "react-router-dom";

export const Home = () => {
    return (
        <div>
            <div className="home">
                <div className="home-text-container">
                    <h2 className="home-header">
                        Welcome to the <span className="page-name">Rick&MortyPedia</span>
                    </h2>
                    <br />
                    <p className="home-paragraph">
                        Here you will find information about characters, episodes they appear on, status, etc.
                        <br />
                    </p>
                    <br />
                    <div className="flex">
                        <a href="https://github.com/danybeltran/react-using-Rick-and-Morty-api" rel="noopener noreferrer" target="_blank" className="github-link">
                            Github
                        </a>
                        <Link to="/characters/page/1" className="home-explore-button">
                            Explore
                        </Link>
                    </div>
                </div>
                <div className="home-image-container">
                    <img src={RickAndMortyTogether} alt="" />
                </div>
            </div>
        </div>
    )
}