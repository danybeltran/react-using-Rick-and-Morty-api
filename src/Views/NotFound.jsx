
import React from "react";
import { Link } from "react-router-dom";

export const NotFound = () => {
    return (
        <div className="not-found">
            <h3 className="error-header">That's a 404 error</h3>
            <p className="error-paragraph">The page you requested doesn't exist.</p>
            <br />
            <Link to="/" className="error-home-link">Go home</Link>
        </div>
    )
}