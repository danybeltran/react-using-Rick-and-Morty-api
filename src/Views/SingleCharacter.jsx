import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import { Helmet } from "react-helmet";

export const SingleCharacter = () => {

    const [character, setCharacter] = useState({
        origin: {},
        location: {},
        episode: []
    });

    const { id } = useParams();

    useEffect(() => {
        const fetchCharacterInformation = async (characterId) => {
            var characterInfo = await fetch("https://rickandmortyapi.com/api/character/" + id);
            var data = await characterInfo.json();
            setCharacter(data);
            console.log(data)
        }
        fetchCharacterInformation(id)
    }, [id]);

    let CharacterStatus = character.status === "Alive" ? "text-green-600" :
        character.status === "Dead" ? "text-red-600" : "text-yellow-600";

    return (
        <div>
            <Helmet>
                <title>
                    {`${character.name} - Rick&MortyPedia`}
                </title>
                <meta property="og:description" content={"Basic information about " + character.name} />
                <meta property="og:image" content={character.image} />
            </Helmet>
            <div className="single-character p-12 md:flex text-gray-700">
                <div className="single-character-image md:w-1/2">
                    <img className="mx-auto md:ml-auto rounded-md shadow-md md:w-1/2 md:mr-3 h-100 object-fill" src={character.image} alt="" />
                </div>
                <br />
                <div className="single-character-information md:px-6">
                    <h4 className="single-character-name text-2xl text-center md:text-left">
                        {character.name}
                    </h4>
                    <p className="single-character-info-paragraph">
                        <span className="font-semibold">{character.species} - <span className={CharacterStatus}>{character.status}</span></span>
                    </p>
                    <p className="single-character-info-paragraph">
                        From: {character.origin.name}
                    </p>
                    <p className="single-character-info-paragraph">
                        Living in: {character.location.name}
                    </p>
                    <br />
                    <p className="single-character-info-paragraph">
                        <span className="text-sm">
                            To date, {character.name} appears in {character.episode.length} {character.episode.length === 1 ? "Episode" : "Episodes"}
                        </span>
                    </p>
                </div>
                <br />
            </div>
        </div>
    )
}