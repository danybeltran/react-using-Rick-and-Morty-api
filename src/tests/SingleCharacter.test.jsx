import React from "react";
import { SingleCharacter } from "../Views/SingleCharacter";

import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

test("Should render information about Rick Sanchez (id 1)", () => {
    const renderdComponent = renderer.create(
        <Router>
            <SingleCharacter testData={
                {
                    name: "Rick Sanchez",
                    image:"https://rickandmortyapi.com/api/character/avatar/1.jpeg",
                    status:"Alive",
                    species:"Human",
                    origin: {
                        name: "Earth"
                    },
                    location: {
                        name: "Earth (Replacement Dimension)"
                    },
                    episode: new Array(41)
                }
            } />
        </Router>
    );

    let component = renderdComponent.toJSON();

    expect(component).toMatchSnapshot();

})