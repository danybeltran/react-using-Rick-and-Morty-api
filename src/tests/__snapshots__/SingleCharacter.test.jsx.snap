// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render information about Rick Sanchez (id 1) 1`] = `
<div>
  <div
    className="single-character p-12 md:flex text-gray-700"
  >
    <div
      className="single-character-image md:w-1/2"
    >
      <img
        alt=""
        className="mx-auto md:ml-auto rounded-md shadow-md md:w-1/2 md:mr-3 h-100 object-fill"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
    </div>
    <br />
    <div
      className="single-character-information md:px-6"
    >
      <h4
        className="single-character-name text-2xl text-center md:text-left"
      >
        Rick Sanchez
      </h4>
      <p
        className="single-character-info-paragraph"
      >
        <span
          className="font-semibold"
        >
          Human
           - 
          <span
            className="text-green-600"
          >
            Alive
          </span>
        </span>
      </p>
      <p
        className="single-character-info-paragraph"
      >
        From: 
        Earth
      </p>
      <p
        className="single-character-info-paragraph"
      >
        Living in: 
        Earth (Replacement Dimension)
      </p>
      <br />
      <p
        className="single-character-info-paragraph"
      >
        <span
          className="text-sm"
        >
          To date, 
          Rick Sanchez
           appears in 
          41
           
          Episodes
        </span>
      </p>
    </div>
    <br />
  </div>
</div>
`;
